components/automate-cli/cmd/chef-automate/verify.go

```diff
@@ -0,0 +1,229 @@
+		"ha-onprem-aws-deploy",
+		false,
+		"Use this flag to verify the On-Premise setup and config with AWS Managed services")
+	verifyCmd.PersistentFlags().BoolVar(
+		&flagsObj.haOnPremCustManagedDeploy,
+		"ha-onprem-customer-deploy",
+		false,
+		"Use this flag to verify the On-Premise setup and config with Customer Managed services")
+	verifyCmd.PersistentFlags().BoolVar(
+		&flagsObj.haAWSDeploy,
+		"ha-aws-deploy",
+		false,
+		"Use this flag to verify the AWS Deployment setup and config")
+	verifyCmd.PersistentFlags().BoolVar(
+		&flagsObj.haAWSManagedDeploy,
+		"ha-aws-managed-deploy",
+		false,
+		"Use this flag to verify the AWS Deployment setup and config with Managed Services")
+	verifyCmd.PersistentFlags().BoolVar(
+		&flagsObj.standaloneDeploy,
+		"standalone-deploy",
+		false,
+		"Use this flag to verify the Automate standalone setup and config")
+	verifyCmd.PersistentFlags().BoolVar(
+		&flagsObj.certificates,
+		"certificates",
+		false,
+		"Use this flag to verify the certificates provided in the file")
+	verifyCmd.PersistentFlags().StringVar(
+		&flagsObj.file,
+		"file",
+		"",
+		"Config file that needs to be verified")
+	RootCmd.AddCommand(verifyCmd)
+}
+
+func verifyCmdFunc(flagsObj *verifyCmdFlags) func(cmd *cobra.Command, args []string) error {
+	return func(cmd *cobra.Command, args []string) error {
+		c := verifyCmdFlow{
+			Verification:      &verification.VerificationModule{},
+			A2HARBFileExist:   isA2HARBFileExist(),
+			ManagedServicesOn: isManagedServicesOn(),
+		}
+		return c.runVerifyCmd(cmd, args, flagsObj)
+	}
+}
+
+func (v *verifyCmdFlow) runVerifyCmd(cmd *cobra.Command, args []string, flagsObj *verifyCmdFlags) error {
+
+	var configPath = ""
+	var err error
+	if len(args) > 0 {
+		configPath = args[0]
+	}
+
+	if flagsObj.haAWSProvision {
+		err = v.verifyHaAWSProvision(configPath)
+	} else if flagsObj.haAWSManagedProvision {
+		err = v.verifyHaAWSManagedProvision(configPath)
+	} else if flagsObj.haAWSDeploy {
+		err = v.verifyHaAWSDeploy(configPath)
+	} else if flagsObj.haAWSManagedDeploy {
+		err = v.verifyHaAWSManagedDeploy(configPath)
+	} else if flagsObj.standaloneDeploy {
+		err = v.verifyStandaloneDeploy(configPath)
+	} else if flagsObj.haOnpremDeploy {
+		err = v.verifyHaOnpremDeploy(configPath)
+	} else if flagsObj.haOnPremAWSManagedDeploy {
+		err = v.verifyHaOnPremAWSManagedDeploy(configPath)
+	} else if flagsObj.haOnPremCustManagedDeploy {
+		err = v.verifyHaOnPremCustManagedDeploy(configPath)
+	} else if flagsObj.certificates {
+		err = v.Verification.VerifyCertificates("")
+	}
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (v *verifyCmdFlow) verifyHaAWSProvision(configPath string) error {
+	if v.A2HARBFileExist {
+		return status.New(status.InvalidCommandArgsError, "Setup is already Provisioned. Please use --ha-aws-deploy flag.")
+	}
+	err := v.Verification.VerifyHAAWSProvision(configPath)
+	if err != nil {
+		return status.Annotate(err, status.ConfigError)
+	}
```

Comment:
above statement can be written in single line statement 
`if err := v.Verification.VerifyHAAWSProvision(configPath); err != nil {
      return status.Annotate(err, status.ConfigError)
 }
`